zkvm/entrypoint/src/lib.rs

```diff
@@ -29,13 +29,21 @@ macro_rules! entrypoint {
     };
 }
 
+pub const PV_DIGEST_NUM_WORDS: usize = 8;
+
 #[cfg(all(target_os = "zkvm", feature = "libm"))]
 mod libm;
 
 #[cfg(target_os = "zkvm")]
 mod zkvm {
     use crate::syscalls::syscall_halt;
+
     use getrandom::{register_custom_getrandom, Error};
+    use once_cell::sync::Lazy;
+    use sha2::{Digest, Sha256};
+    use std::sync::Mutex;
+
+    pub static PUBLIC_VALUES_HASHER: Lazy<Mutex<Sha256>> = Lazy::new(|| Mutex::new(Sha256::new()));
```

Comment:
since the zkvm is singlethreaded I don't think a mutex is necessary, we can assume we will access it safely. Maybe just Lazy<Sha256> + `static mut` will work, which is like what is done in sys_alloc_aligned